arch/arm64/core/switch.S

```diff
@@ -40,9 +42,9 @@ SECTION_FUNC(TEXT, z_arm64_context_switch)
 	stp	x27, x28, [x1, #_thread_offset_to_callee_saved_x27_x28]
 	stp	x29, x4,  [x1, #_thread_offset_to_callee_saved_x29_sp_el0]
 
-	/* Save the current SP_ELx */
+	/* Save the current SP_ELx and return address */
 	mov	x4, sp
-	str	x4, [x1, #_thread_offset_to_callee_saved_sp_elx]
+	stp	x4, lr, [x1, #_thread_offset_to_callee_saved_sp_elx_lr]
```

Comment:
nit: can we keep using `x30` instead of `lr` to be consistent with the current code?